src/plugins/MediaManager.js

```diff
@@ -0,0 +1,341 @@
+                height: this.videoHeight,
+            },
+        })
+            .catch((err) => {
+                this.onVideoStreamError(err);
+            })
+            .then((mediaStream) => {
+                log.debug('Client video stream ready');
+                this.videoStreaming = true;
+                this.localVideoStream = mediaStream;
+                this.addVideoStream(mediaStream);
+            });
+    }
+
+    /**
+     * Initialize and start client microphone audio stream.
+     */
+    startAudioStreaming() {
+        if (!navigator.mediaDevices) {
+            return;
+        }
+
+        navigator.mediaDevices.getUserMedia({
+            audio: true,
+            video: false
+        })
+            .catch((err) => {
+                this.onAudioStreamError(err);
+            })
+            .then((mediaStream) => {
+                log.debug('Client audio stream ready');
+                this.audioStreaming = true;
+                this.localAudioStream = mediaStream;
+                this.addAudioStream(mediaStream);
+            });
+    }
+
+    /**
+     * Client video stream error handler.
+     *
+     * @param {Error} error Camera stream error.
+     */
+    onVideoStreamError(error) {
+        log.warn('Can\'t start client video stream', error);
+    }
+
+    /**
+     * Client audio stream error handler.
+     *
+     * @param {Error} error Microphone stream error.
+     */
+    onAudioStreamError(error) {
+        log.warn('Can\'t start client audio stream', error);
+    }
+
+    /**
+     * Stop client webcam video stream.
+     */
+    stopVideoStreaming() {
+        log.debug('removed local video stream');
+        this.removeVideoStream(this.localVideoStream);
+        this.localVideoStream = null;
+        this.videoStreaming = false;
+    }
+
+    /**
+     * Stop client microphone audio stream.
+     */
+    stopAudioStreaming() {
+        log.debug('removed local audio stream');
+        this.removeAudioStream(this.localAudioStream);
+        this.localAudioStream = null;
+        this.audioStreaming = false;
+    }
+
+    /**
+     * Listener to the camera permission change. Reemits a custom event dispatched on the window
+     */
+    onCameraPermissionChange() {
+        window.dispatchEvent(new CustomEvent('gm-cameraPermissionChange'));
+    }
+
+    /**
+     * Listener to the microphone permission change. Reemits a custom event dispatched on the window
+     */
+    onMicrophonePermissionChange() {
+        window.dispatchEvent(new CustomEvent('gm-microphopnePermissionChange'));
+    }
+
+    /**
+     * Find RTCRtpSender corresponding RTCRtpTransceiver and set its direction.
+     *
+     * @param {RTCRtpSender} sender    Peer (sender).
+     * @param {string}       direction Transceiver direction.
+     */
+    setTransceiverDirection(sender, direction) {
+        // find transceiver that contains sender
+        this.instance.peerConnection.getTransceivers().forEach((transceiver) => {
```

Comment:
don't know if you can get multiple times the same sender but if not you can use find / findIndex, it's less clearer but the loop stop after getting the first row which match

const arrTransceivers =  this.instance.peerConnection.getTransceivers()
const index = arrTransceivers .findIndex(transceiver => transceiver.sender === sender)
if (index > -1)  arrTransceivers[index].direction = direction

Reply:
This needs to be verified as I am not sure if there was multiple instance of the same sender when I did this dev

Reply:
I admit I'm not sure what's the point of setting the direction in the first place. I guess if the transceiver is in the `receive` direction we can't send data through the stream ? But how/why would it be in the `receive` direction?

Reply:
(since I don't understand the aim here, I'm not sure how I can verify that @jparez's suggestion doesn't break stuff)

Reply:
closing this suggestion since the existing code works already and I don't know how to check that I'm not breaking stuff by changing it